namer/consul/src/main/scala/io/buoyant/namer/consul/DcServices.scala

```diff
@@ -62,7 +62,14 @@ private[consul] object DcServices {
             // may consider retrying certain failures (with backoff).
             state() = Activity.Failed(e)
             stats.errors.incr()
-            Future.exception(e)
+            log.debug("consul connection failed: %s", name)
+
+            e match {
```

Comment:
hmmmm... strange.  The only thing I can think of is that RetryFilter will skip any Failures that are flagged `f.isFlagged(FailureFlags.NonRetryable)` without consulting the retry policy.  Maybe the connection failed exceptions are flagged this way?  Not sure why they would be though....